app/models/coin.rb

```diff
@@ -149,12 +149,19 @@ def live_market_data
     Rails.cache.fetch("coins/#{id}/market_data", expires_in: 1.minute) do
       url = "https://api.coinmarketcap.com/v1/ticker/#{slug}/?convert=BTC"
       response = HTTParty.get(url)
-      data = JSON.parse(response.body)[0] || {}
-      default_market_data.merge(data)
+      if response.code == 200
```

Comment:
Is it better to use .success? Also, seems possible for the response body to have a body.metadata.error as well.

Reply:
Good call, we could get non-200 responses that are still valid.

Good catch on `body` potentially having an error as well, but the only documented way for that to happen is via bad input parameters. It's definitely possible for CMC to just return something invalid but your work on CMC Pro will be replacing this soon, this is just a bandaid until that's released so I'd prefer to skip that level of error handling for now.